pkg/kubelet/eviction/helpers.go

```diff
@@ -0,0 +1,155 @@
+	softThresholds, err := parseThresholdStatements(evictionSoft)
+	if err != nil {
+		return nil, err
+	}
+	gracePeriods, err := parseGracePeriods(evictionSoftGracePeriod)
+	if err != nil {
+		return nil, err
+	}
+	for i := range softThresholds {
+		signal := softThresholds[i].Signal
+		period, found := gracePeriods[signal]
+		if !found {
+			return nil, fmt.Errorf("grace period must be specified for the soft eviction threshold %v", signal)
+		}
+		softThresholds[i].GracePeriod = period
+	}
+	results = append(results, softThresholds...)
+	return results, nil
+}
+
+// parseThresholdStatements parses the input statements into a list of Threshold objects.
+func parseThresholdStatements(expr string) ([]Threshold, error) {
+	if len(expr) == 0 {
+		return nil, nil
+	}
+	results := []Threshold{}
+	statements := strings.Split(expr, ",")
+	for _, statement := range statements {
+		result, err := parseThresholdStatement(statement)
+		if err != nil {
+			return nil, err
+		}
+		results = append(results, result)
+	}
+	return results, nil
+}
+
+// parseThresholdStatement parses a threshold statement.
+func parseThresholdStatement(statement string) (Threshold, error) {
+	tokens2Operator := map[string]ThresholdOperator{
+		"<": OpLessThan,
+		">": OpGreaterThan,
+	}
+	var (
+		operator ThresholdOperator
+		parts    []string
+	)
+	for token := range tokens2Operator {
+		parts = strings.Split(statement, token)
+		// if we got a token, we know this was the operator...
+		if len(parts) > 1 {
+			operator = tokens2Operator[token]
+			break
+		}
+	}
+	if len(operator) == 0 || len(parts) != 2 {
+		return Threshold{}, fmt.Errorf("invalid eviction threshold syntax %v, expected <signal><operator><value>", statement)
+	}
+	signal := Signal(parts[0])
+	if !validSignal(signal) {
+		return Threshold{}, fmt.Errorf("unsupported eviction signal %v", signal)
+	}
+
+	quantity, err := resource.ParseQuantity(parts[1])
+	if err != nil {
+		return Threshold{}, err
+	}
+	return Threshold{
+		Signal:   signal,
+		Operator: operator,
+		Value:    *quantity,
+	}, nil
+}
+
+// parseGracePeriods parses the grace period statements
+func parseGracePeriods(expr string) (map[Signal]time.Duration, error) {
+	if len(expr) == 0 {
+		return nil, nil
+	}
+	results := map[Signal]time.Duration{}
+	statements := strings.Split(expr, ",")
+	for _, statement := range statements {
+		parts := strings.Split(statement, "=")
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("invalid eviction grace period syntax %v, expected <signal>=<duration>", statement)
+		}
+		signal := Signal(parts[0])
+		if !validSignal(signal) {
+			return nil, fmt.Errorf("unsupported eviction signal %v", signal)
+		}
+
+		gracePeriod, err := time.ParseDuration(parts[1])
+		if err != nil {
+			return nil, err
+		}
+		if gracePeriod < 0 {
+			return nil, fmt.Errorf("invalid eviction grace period specified: %v, must be a positive value", parts[1])
+		}
+
+		// check against duplicate statements
+		if _, found := results[signal]; found {
```

Comment:
I don't think so, we could check that if you have:

hard=memory.available<X
soft=memory.available<Y

We could verify that X > Y, but not sure its worth the complication if we expose more operators.